src/transformers/testing_utils.py

```diff
@@ -1570,6 +1572,34 @@ def LoggingLevel(level):
         transformers_logging.set_verbosity(orig_level)
 
 
+class TemporaryHubRepo:
+    """
+    This is a context manager to temporarily change transformers modules logging level to the desired value and have it
+    restored to the original setting at the end of the scope.
+
+    Example:
+
+    ```python
+    with LoggingLevel(logging.INFO):
+        AutoModel.from_pretrained("openai-community/gpt2")  # calls logger.info() several times
+    ```
+    """
+    def __init__(self, prefix="repo", random_id=None, token=None):
+        self.prefix = prefix
+        self.random_id = random_id
+        self.token = token
+        self.repo_id = f"{self.prefix}-{self.random_id}"
+
+    def __enter__(self):
+        return self.repo_id
+
+    def __exit__(self, exc, value, tb):
+        try:
+            delete_repo(repo_id=self.repo_id, token=self.token)
+        except huggingface_hub.utils.RepositoryNotFoundError as e:
+            pass
```

Comment:
```suggestion
        delete_repo(repo_id=self.repo_id, token=self.token, missing_ok=True)
```

You can also pass `missing_ok=True` as parameter (will catch the repo not found error for you)